scripts/objectstore/copy_files_to_irods.py

```diff
@@ -0,0 +1,366 @@
+    disk_file_checksum = subprocess.check_output(checksum_cmd, shell=True)
+    # remove '\n' from the end of disk_file_checksum
+    disk_file_checksum_len = len(disk_file_checksum)
+    disk_file_checksum_trimmed = disk_file_checksum[0:(disk_file_checksum_len - 1)]
+    # Return Unicode string
+    return disk_file_checksum_trimmed.decode("utf-8")
+
+
+def print_help_msg():
+    print("\nThis script copies datasets from Disk to Irods. It operates on a range of dataset IDs (start_dataset_id and end_dtatset_id, both inclusive).")
+    print("It also takes in the name of 3 JSON configuration files (A sample JSON configuration file is provided in the Python script folder).")
+    print("\n1) object_store_info_file: Has Disk object store info (e.g. files info).")
+    print("2) irods_info_file: irods connection info, and misc. info (zone, home, obejct store ID, etc.).")
+    print("3) db_connection_info: Galaxy DB connection info.")
+    print("\nIt also takes a flag copy_or_checksum.")
+    print("If copy: for all datasets from start_dataset_it to end_dataset_id, it copies disk files to irods. This includes extra folders that come with the output of commands like fastqc.")
+    print("If checksum: for all datasets from start_dataset_id to end_dataset_id, it calculates the disk file checksum, gets the irods file, and compares the disk file checksum to irods file checksum.")
+    print("If checksums match, it updates object_store_id dataset table (so Galaxy starts using the irods version of the file), then DELETES the file from disk.")
+    print("If the checksumdoes NOT match: it prints and error message, does NOT update the dataset table or delete the disk file, and continues with the next dataset in dataset ID range.")
+    print("\nIn order to run this Python3 script, you need to install the following packages via pip3:")
+    print("\tpip3 install galaxy.util")
+    print("\tpip3 install psycopg2")
+    print("\tpip3 install python_irodsclient")
+    print("\nLong-form input parameter specification:")
+    print("copy_files_to_irods --start_dataset_id=2 --end_dataset_id=3 --object_store_info_file=object_store_info.json --irods_info_file=irods_info.json --db_connection_info_file=db_connection_info.json --copy_or_checksum=<copy|checksum>")
+    print("\nOR")
+    print("\nShort-form input parameter specification:")
+    print("copy_files_to_irods -s 2 -e 3 -o object_store_info.json -i irods_info.json -d db_connection_info.json -c <copy|checksum>")
+    print("\n")
+
+
+if __name__ == '__main__':
+    start_dataset_id = None
+    end_dataset_id = None
+    object_store_info_file = None
+    irods_info_file = None
+    db_connection_info_file = None
+    copy_or_checksum = None
+
+    try:
+        opts, args = getopt.getopt(sys.argv[1:], "hs:e:o:i:d:c:", ["start_dataset_id=", "end_dataset_id=", "object_store_info_file=", "irods_info_file=", "db_connection_info_file=", "copy_or_checksum="])
```

Comment:
Can you use argparse here for a friendlier interface ?

Reply:
Sure.